plugins/filter_aws/aws.c

```diff
@@ -370,6 +400,264 @@ static int get_vpc_metadata(struct flb_filter_aws *ctx)
     return ret;
 }
 
+static void get_cluster_from_environment(struct flb_filter_aws *ctx)
+{
+    if(ctx->cluster == NULL) {
+        char* cluster_name = getenv("CLUSTER_NAME");
+        if(cluster_name) {
+            ctx->cluster = strdup(cluster_name);
+            ctx->cluster_len = strlen(cluster_name);
+            ctx->new_keys++;
+        } else {
+            free(cluster_name);
+        }
+        flb_plg_debug(ctx->ins, "Cluster name is %s.", ctx->cluster);
+    }
+}
+
+static int file_to_buffer(const char *path,
+                          char **out_buf, size_t *out_size)
+{
+    int ret;
+    char *buf;
+    ssize_t bytes;
+    FILE *fp;
+    struct stat st;
+
+    if (!(fp = fopen(path, "r"))) {
+        return -1;
+    }
+
+    ret = stat(path, &st);
+    if (ret == -1) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+
+    buf = flb_calloc(1, (st.st_size + 1));
+    if (!buf) {
+        flb_errno();
+        fclose(fp);
+        return -1;
+    }
+
+    bytes = fread(buf, st.st_size, 1, fp);
+    if (bytes < 1) {
+        flb_free(buf);
+        fclose(fp);
+        return -1;
+    }
+
+    fclose(fp);
+
+    *out_buf = buf;
+    *out_size = st.st_size;
+
+    return 0;
+}
+
+/* Set K8s Authorization Token and get HTTP Auth Header */
+static int get_http_auth_header(struct flb_filter_aws *ctx) 
+{
+    int ret;
+    char *temp;
+    char *tk = NULL;
+    size_t tk_size = 0;
+
+    ret = file_to_buffer(FLB_KUBE_TOKEN, &tk, &tk_size);
+    if (ret == -1) {
+        flb_plg_warn(ctx->ins, "cannot open %s", FLB_KUBE_TOKEN);
+    }
+    flb_plg_info(ctx->ins, " token updated");
```

Comment:
nit: this log line is vague. Could just follow the original log line pasted from `kube_meta.c`
```suggestion
    flb_plg_info(ctx->ins, " token updated", FLB_KUBE_TOKEN);
```

Reply:
Will update in revision